tests/integration/test_sidecar_upgrade.py

```diff
@@ -167,13 +177,22 @@ async def build_and_upgrade_fixture(
     charm = await ops_test.build_charm(".")
     # Most of the times the integration tests will fail due to timeout without the force flag.
     app: juju.application.Application = ops_test.model.applications[application_name]
+    wordpress_revision_before = await get_app_revision(application_name)
+    last_revision_using_ingress = 14
+    upgrade_from_ingress = wordpress_revision_before <= last_revision_using_ingress
+    if upgrade_from_ingress:
```

Comment:
This might start to become unmanageable as we keep changing the wordpress charm. Should we add specific scenarios for specific revisions that make use of common fixtures?

Reply:
This is not the norm of the WordPress upgrade, as we would like to provide smooth upgrade experience for users. WordPress upgrade requires additional steps like this is mostly one time thing and I imagine it would happen very often.